### PR TITLE
feat: add support for PSC

### LINKED ISSUE #290: Add support for PSC Connections
Add a new IP type `IPTypes.PSC` to support connections using [private service connect](https://cloud.google.com/vpc/docs/private-service-connect).

### PR DESCRIPTION
**Pull request by ttosta-google**
Add support to Python Connector for PSC connections:

- [x] Code changes
- [x] Add unit tests
- [x] Add integration tests

Fixes #290

### CODE REVIEW

File: google/cloud/alloydb/connector/client.py [hunk 1]
```diff
@@ -129,9 +129,15 @@ async def _get_metadata(
         resp = await self._client.get(url, headers=headers, raise_for_status=True)
         resp_dict = await resp.json()
 
+        # Remove trailing period from PSC DNS name.
+        psc_dns = resp_dict.get("pscDnsName")
+        if psc_dns:
+            psc_dns = psc_dns.rstrip(".")
```
Comment (enocom) [inline google/cloud/alloydb/connector/client.py:135]: Was this necessary in Python?

File: tests/system/test_asyncpg_psc.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Tuple
+
+# [START alloydb_sqlalchemy_connect_async_connector_psc]
```
Comment (enocom) [inline tests/system/test_asyncpg_psc.py:18]: We can probably remove these region tags now that we've settled on just showing a few lines for PSC instead in the public docs.

**Review by enocom — APPROVED** [review]
@jackwotherspoon Mind taking a look as well?

File: google/cloud/alloydb/connector/client.py (hunk 1 above)
Comment (ttosta-google) [reply google/cloud/alloydb/connector/client.py:135]: I tried without it and it failed: 

```
>       self._sslobj.do_handshake()
E       ssl.SSLError: [SSL: SSLV3_ALERT_UNEXPECTED_MESSAGE] sslv3 alert unexpected message (_ssl.c:1131)
```

https://github.com/GoogleCloudPlatform/alloydb-python-connector/actions/runs/8471675334/job/23489146452

File: tests/system/test_asyncpg_psc.py (hunk 2 above)
Comment (ttosta-google) [reply tests/system/test_asyncpg_psc.py]: removed.

**Review by jackwotherspoon — APPROVED** [review]
LGTM, awesome job!

**Pull request merged**